lib/hal_api/controller/filtering.rb

```diff
@@ -53,6 +53,24 @@ def filters
     @filters ||= parse_filters_param
   end
 
+  def filter_facets
+  end
+
+  def index_collection
+    collection = defined?(super) ? super : HalApi::PagedCollection.new([])
+
+    # add facets if defined, removing filters/facets with counts of 0
+    non_zero_facets = (filter_facets || {}).with_indifferent_access.tap do |hash|
```

Comment:
Should the filter_key have more than just an id? Should there be a human readable label for it?
e.g. `some_custom_filter` would not be the best for display.

Reply:
I have talked myself out of this, withdrawn!

Reply:
I'd lump this in with "describing _all available_ filters", which is sort of out-of-scope for this PR now.  Opened #19.